packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -326,6 +326,18 @@ class IosProject extends XcodeBasedProject {
     await _updateGeneratedXcodeConfigIfNeeded();
   }
 
+  String? xcworkspacePath() {
+    final List<FileSystemEntity> contents = hostAppRoot.listSync();
+    for (final FileSystemEntity entity in contents) {
+      // On certain volume types, there is sometimes a stray `._Runner.xcworkspace` file.
+      // Find the first non-hidden xcworkspace and return the path.
```

Comment:
ignoring hidden files, are there sometimes multiple xcworkspace files here? If so, does it matter which we pick?

Reply:
I don't believe there should be other xcworkspace files. @jmagman What do you think?

Reply:
No, there shouldn't be.  And it was picking the first one found before this PR, and this still does the same.